hive-plugins/src/main/java/co/cask/plugin/etl/batch/sink/HiveBatchSink.java

```diff
@@ -0,0 +1,142 @@
+ * the License.
+ */
+
+package co.cask.plugin.etl.batch.sink;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.batch.OutputFormatProvider;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.api.dataset.lib.KeyValueTable;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.BatchRuntimeContext;
+import co.cask.cdap.etl.api.batch.BatchSink;
+import co.cask.cdap.etl.api.batch.BatchSinkContext;
+import co.cask.plugin.etl.batch.commons.HiveSchemaConverter;
+import co.cask.plugin.etl.batch.commons.HiveSchemaStore;
+import com.google.common.base.Objects;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.WritableComparable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hive.hcatalog.data.DefaultHCatRecord;
+import org.apache.hive.hcatalog.data.HCatRecord;
+import org.apache.hive.hcatalog.data.schema.HCatSchema;
+import org.apache.hive.hcatalog.mapreduce.HCatOutputFormat;
+import org.apache.hive.hcatalog.mapreduce.OutputJobInfo;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Hive Batch Sink
+ */
+@Plugin(type = "batchsink")
+@Name("Hive")
+@Description("Batch Sink to write to external Hive tables.")
+public class HiveBatchSink extends BatchSink<StructuredRecord, NullWritable, HCatRecord> {
+
+  private static final String DEFAULT_HIVE_DATABASE = "default";
+  private static final Gson GSON = new Gson();
+  private static final Type STRING_MAP_TYPE = new TypeToken<Map<String, String>>() {
+  }.getType();
+
+  private HiveSinkConfig config;
+  private RecordToHCatRecordTransformer recordToHCatRecordTransformer;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    //TODO: remove this way of storing Hive schema once we can share info between prepareRun and initialize stage.
+    pipelineConfigurer.createDataset(HiveSchemaStore.HIVE_TABLE_SCHEMA_STORE, KeyValueTable.class,
+                                     DatasetProperties.EMPTY);
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    Job job = context.getHadoopJob();
+    job.setOutputKeyClass(WritableComparable.class);
+    job.setOutputValueClass(DefaultHCatRecord.class);
+    Configuration configuration = job.getConfiguration();
+    configuration.set("hive.metastore.uris", config.metaStoreURI);
+    HCatOutputFormat.setOutput(configuration, job.getCredentials(), OutputJobInfo.create(Objects.firstNonNull(
```

Comment:
shouldn't use this method, but should make sure all the settings this sets are reflected in the output format provider's getOutputFormatConfiguration method.  Otherwise this will fail when writing to multiple sinks.